### PR TITLE
[types] overload multiplyMatrices types

### PR DESCRIPTION
**Pull request by epsilonError**
Adds overloads for the parameter and return types of multiplyMatrices. And tests in `types/test` for the overloads.

Includes updates to `adapt.js` and `CATs.js` to fix type errors due to the new overloaded types.

~I was probably a bit too overzealous and added functionality for returning a number result as well. 
Let me know if this new functionality should be in a separate pull request.~ Removed to add in future.

@MysteryBlokHed, hopefully I didn't step on any in-progress type fixes.

### CODE REVIEW

**Comment by MysteryBlokHed** [comment]
> @MysteryBlokHed, hopefully I didn't step on any in-progress type fixes.

Luckily, the PR we had to fix a handful of type errors (#574) was merged a couple days before this was opened :)

I'll try to get this PR reviewed before tomorrow, if possible.

**Comment by lloydk** [comment]
> Adds overloads for the parameter and return types of multiplyMatrices. And tests in `types/test` for the overloads.
> 
> Includes updates to `adapt.js` and `CATs.js` to fix type errors due to the new overloaded types.
> 
> I was probably a bit too overzealous and added functionality for returning a number result as well. Let me know if this new functionality should be in a separate pull request.
> 
> @MysteryBlokHed, hopefully I didn't step on any in-progress type fixes.

**Comment by lloydk** [comment]
Oops

File: src/adapt.js
Comment (lloydk) [reply src/adapt.js:73]: I wouldn't worry about it for now. If it becomes an issue in the future we can deal with it then.

**Review by lloydk — COMMENTED** [review]
>I was probably a bit too overzealous and added functionality for returning a number result as well.
Let me know if this new functionality should be in a separate pull request.

I think this should be a separate PR with tests for the new functionality (probably want to wait until #559 is merged).

Everything else LGTM

**Review by MysteryBlokHed — APPROVED** [review]
Overload changes LGTM

**Review by lloydk — APPROVED** [review]

**Pull request merged**
